### PR TITLE
[changelog-json] The extension supports the encode.decimal-as-plain-number parameter #issue-527

### PR DESCRIPTION
**Pull request by liaooo**
将changelog-json内部依赖的flink-json，创建JsonRowDataSerializationSchema实例时使用了encode.decimal-as-plain-number默认值。本次修改使该选项放开到changelog-json的选项可配置，以解决在使用了Decimal类型字段场景中会序列化为科学计数法的问题。

### CODE REVIEW

File: flink-format-changelog-json/src/main/java/com/ververica/cdc/formats/json/ChangelogJsonFormatFactory.java [hunk 1]
```diff
@@ -108,7 +112,8 @@ public ChangelogMode getChangelogMode() {
             public SerializationSchema<RowData> createRuntimeEncoder(
                     DynamicTableSink.Context context, DataType consumedDataType) {
                 final RowType rowType = (RowType) consumedDataType.getLogicalType();
-                return new ChangelogJsonSerializationSchema(rowType, timestampFormat);
+                return new ChangelogJsonSerializationSchema(
+                        rowType, timestampFormat, encodeDecimalAsPlainNumber);
```
Comment (mincwang) [inline flink-format-changelog-json/src/main/java/com/ververica/cdc/formats/json/ChangelogJsonFormatFactory.java:116]: For better extension, Can use the `new ChangelogJsonSerializationSchema(rowType, timestampFormat, formatOptions)` ?

Comment (liaooo) [reply flink-format-changelog-json/src/main/java/com/ververica/cdc/formats/json/ChangelogJsonFormatFactory.java:116]: I modified the constructor at your suggestion, Please review the code again.

File: flink-format-changelog-json/src/test/java/com/ververica/cdc/formats/json/ChangelogJsonSerDeDecimalTest.java [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+        List<String> expected =
+                Arrays.asList(
+                        "+I(112,liaooo,10.00)",
+                        "-U(112,liaooo,10.00)",
+                        "+U(112,liaooo,1000000.00)",
+                        "-U(112,liaooo,1000000.00)",
+                        "+U(112,liaooo,9999999.50)");
+        List<String> actual =
+                collector.list.stream().map(Object::toString).collect(Collectors.toList());
+        assertEquals(expected, actual);
```
Comment (mincwang) [inline flink-format-changelog-json/src/test/java/com/ververica/cdc/formats/json/ChangelogJsonSerDeDecimalTest.java:85]: We should add a deserialization test data for scientific notation？

File: flink-format-changelog-json/src/test/java/com/ververica/cdc/formats/json/ChangelogJsonSerDeDecimalTest.java [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+
+        Configuration formatOptions = new Configuration();
+        formatOptions.setString(JsonOptions.TIMESTAMP_FORMAT.key(), JsonOptions.SQL);
+        formatOptions.setBoolean(JsonOptions.ENCODE_DECIMAL_AS_PLAIN_NUMBER.key(), false);
+
+        // decimal字段会启用科学记数法
```
Comment (mincwang) [inline flink-format-changelog-json/src/test/java/com/ververica/cdc/formats/json/ChangelogJsonSerDeDecimalTest.java:91]: pelase use English commonts

File: flink-format-changelog-json/src/test/java/com/ververica/cdc/formats/json/ChangelogJsonSerDeDecimalTest.java [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+        List<String> expectedResult =
+                Arrays.asList(
+                        "{\"data\":{\"id\":112,\"name\":\"liaooo\",\"money\":1E+1},\"op\":\"+I\"}",
+                        "{\"data\":{\"id\":112,\"name\":\"liaooo\",\"money\":1E+1},\"op\":\"-U\"}",
+                        "{\"data\":{\"id\":112,\"name\":\"liaooo\",\"money\":1E+6},\"op\":\"+U\"}",
+                        "{\"data\":{\"id\":112,\"name\":\"liaooo\",\"money\":1E+6},\"op\":\"-U\"}",
+                        "{\"data\":{\"id\":112,\"name\":\"liaooo\",\"money\":9999999.5},\"op\":\"+U\"}");
+        assertEquals(expectedResult, result);
+
+        // 使decimal字段以plain形式展示
```
Comment (mincwang) [inline flink-format-changelog-json/src/test/java/com/ververica/cdc/formats/json/ChangelogJsonSerDeDecimalTest.java:108]: ditto

File: flink-format-changelog-json/src/test/java/com/ververica/cdc/formats/json/ChangelogJsonSerDeDecimalTest.java [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+                    new String(serializationSchema2.serialize(rowData), StandardCharsets.UTF_8));
+        }
+        List<String> expectedResult2 =
+                Arrays.asList(
+                        "{\"data\":{\"id\":112,\"name\":\"liaooo\",\"money\":10},\"op\":\"+I\"}",
+                        "{\"data\":{\"id\":112,\"name\":\"liaooo\",\"money\":10},\"op\":\"-U\"}",
+                        "{\"data\":{\"id\":112,\"name\":\"liaooo\",\"money\":1000000},\"op\":\"+U\"}",
+                        "{\"data\":{\"id\":112,\"name\":\"liaooo\",\"money\":1000000},\"op\":\"-U\"}",
+                        "{\"data\":{\"id\":112,\"name\":\"liaooo\",\"money\":9999999.5},\"op\":\"+U\"}");
+        assertEquals(expectedResult2, result2);
```
Comment (mincwang) [inline flink-format-changelog-json/src/test/java/com/ververica/cdc/formats/json/ChangelogJsonSerDeDecimalTest.java:125]: A separate serialization method should be extracted here

**Review by mincwang — COMMENTED** [review]
Thanks @liaooo for the contribution, i left a few comments.

File: flink-format-changelog-json/src/test/java/com/ververica/cdc/formats/json/ChangelogJsonSerDeDecimalTest.java (hunk 2 above)
Comment (liaooo) [reply flink-format-changelog-json/src/test/java/com/ververica/cdc/formats/json/ChangelogJsonSerDeDecimalTest.java:87]: i refined the relevant test cases and comments, review again.

**Review by mincwang — APPROVED** [review]
LGTM

**Comment by leonardBang** [comment]
I'm sorry for missed this ticket @liaooo @mincwang, we have remove #901  and recommend user to use other well-supported CDC json format, thus this PR will be deprecated, sorry again for my miss.

**Comment by liaooo** [comment]
> I'm sorry for missed this ticket @liaooo @mincwang, we have remove #901 and recommend user to use other well-supported CDC json format, thus this PR will be deprecated, sorry again for my miss.

OK！

**Comment by ruanhang1993** [comment]
cc @leonardBang Maybe we could close this PR.

**Pull request closed without merging**